pkgs/development/zig-packages/build-zig-package/default.nix

```
@@ -0,0 +1,58 @@
+{ lib, stdenv, zig, mktemp }:
+
+{ strictDeps ? true
+, depsBuildBuild ? [ ]
+, nativeBuildInputs ? [ ]
+, configurePhase ? null
+, buildPhase ? null
+, checkPhase ? null
+, installPhase ? null
+, enableParallelBuilding ? true
+, meta ? { }
+, ...
+}@attrs:
+
+stdenv.mkDerivation (attrs // {
+  inherit strictDeps enableParallelBuilding;
+  depsBuildBuild = [ ] ++ depsBuildBuild;
+  nativeBuildInputs = [ zig ] ++ nativeBuildInputs;
+
+  configurePhase =
+    if (configurePhase == null) then ''
+      runHook preConfigure
+      runHook postConfigure
+    '' else
+      configurePhase;
+
+  postConfigure = ''
+    export XDG_CACHE_HOME=$(mktemp -d)
+  '';
+
+  buildPhase =
+    if (buildPhase == null) then ''
+      runHook preBuild
+      zig build --verbose
+      runHook postBuild
+    '' else
+      buildPhase;
+
+  checkPhase =
+    if (checkPhase == null) then ''
+      runHook preCheck
+      runHook postCheck
+    '' else
+      checkPhase;
+
+  installPhase =
+    if (installPhase == null) then ''
+      runHook preInstall
+      zig build --prefix $out install
```

```suggestion
      zig build -Drelease-safe -Dcpu=baseline --prefix $out install
```

We ideally want binaries in release mode, and `-Dcpu=baseline` to ensure reproducibility because Zig does CPU native builds by default.

---

Okay, I saw this in a zig package build in nixpkgs. But I didn't know if it was specific to that package or not.